tests/rkt_fetch_test.go

```diff
@@ -168,3 +174,180 @@ func testFetchNoStore(t *testing.T, args string, image string, finalURL string)
 	}
 	child.Wait()
 }
+
+func TestResumedFetch(t *testing.T) {
+	image := "rkt-inspect-implicit-fetch.aci"
+	imagePath := patchTestACI(image, "--exec=/inspect")
+
+	hash := types.ShortHash("sha512-" + getHashOrPanic(imagePath))
+
+	defer os.Remove(imagePath)
+
+	kill := make(chan struct{})
+	reportkill := make(chan struct{})
+
+	server := httptest.NewServer(testInterruptingServerHandler(t, imagePath, kill, reportkill))
+	defer server.Close()
+
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	cmd := fmt.Sprintf("%s --no-store --insecure-skip-verify fetch %s", ctx.Cmd(), server.URL)
+	child := spawnOrFail(t, cmd)
+	<-kill
+	err := child.Close()
+	if err != nil {
+		panic(err)
+	}
+	reportkill <- struct{}{}
+
+	// rkt has fetched the first half of the image
+	// If it fetches the first half again these channels will be written to.
+	// Closing them to make the test panic if they're written to.
+	close(kill)
+	close(reportkill)
+
+	child = spawnOrFail(t, cmd)
+	if _, _, err := expectRegexWithOutput(child, ".*"+hash); err != nil {
+		t.Fatalf("hash didn't match: %v", err)
+	}
+	err = child.Wait()
+	if err != nil {
+		t.Errorf("rkt didn't exit cleanly: %v", err)
+	}
+}
+
+func TestResumedFetchInvalidCache(t *testing.T) {
+	image := "rkt-inspect-implicit-fetch.aci"
+	imagePath := patchTestACI(image, "--exec=/inspect")
+	defer os.Remove(imagePath)
+
+	hash := types.ShortHash("sha512-" + getHashOrPanic(imagePath))
+
+	kill := make(chan struct{})
+	reportkill := make(chan struct{})
+
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	// Fetch the first half of the image, and kill rkt once it reaches halfway.
+	server := httptest.NewServer(testInterruptingServerHandler(t, imagePath, kill, reportkill))
+	defer server.Close()
+	cmd := fmt.Sprintf("%s --no-store --insecure-skip-verify fetch %s", ctx.Cmd(), server.URL)
+	child := spawnOrFail(t, cmd)
+	<-kill
+	err := child.Close()
+	if err != nil {
+		panic(err)
+	}
+	reportkill <- struct{}{}
+
+	// Fetch the image again. The server doesn't support Etags or the
+	// Last-Modified header, so the cached version should be invalidated. If
+	// rkt tries to use the cache, the hash won't check out.
+	server2 := httptest.NewServer(testSimpleServerHandler(t, imagePath))
+	defer server2.Close()
+	cmd = fmt.Sprintf("%s --no-store --insecure-skip-verify fetch %s", ctx.Cmd(), server2.URL)
```

Comment:
`server.URL` might not be the same as `server2.URL`, so rkt might not even try to use the cache.